CONTRIBUTING/index.html

```diff
@@ -1,4168 +1,10 @@
-
-<!doctype html>
-<html lang="en" class="no-js">
+<html xmlns="http://www.w3.org/1999/xhtml">
   <head>
-    
-      <meta charset="utf-8">
-      <meta name="viewport" content="width=device-width,initial-scale=1">
-      
-      
-      
-      <link rel="icon" href="../assets/images/favicon.png">
-      <meta name="generator" content="mkdocs-1.2.4, mkdocs-material-8.1.9">
-    
-    
-      
-        <title>Contributing - Argo Workflows - The workflow engine for Kubernetes</title>
-      
-    
-    
-      <link rel="stylesheet" href="../assets/stylesheets/main.2b4465f4.min.css">
-      
-        
-        <link rel="stylesheet" href="../assets/stylesheets/palette.e6a45f82.min.css">
-        
-      
-    
-    
-    
-      
-        
-        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
-        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
-        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
```

Comment:
So I did mention this a tiny bit in https://github.com/argoproj/argo-workflows/pull/12212#issuecomment-1816674293, I think it would be good to leave some styling in so it looks a little less janky/unofficial to users.

We could add more in follow-up PRs, but as you left in the `assets/` dir, we can continue using the CSS and icons from there.

Otherwise we may want to delete or redirect the no longer used JS and images from this branch.

Reply:
Sure re: styling but the redirect should work.  I think it is less work for us to manage styling for those people who it doesn't redirect for.  Keeping it simple is better IMO.

I'm good removing all other files also as part of this PR.  People should not be viewing the assets (css, js, etc) directly anyway IMO.

wdyt?

Reply:
> I think it is less work for us to manage styling for those people who it doesn't redirect for.

It's not just for those who the redirect fails for, it's also for the initial load; since we can't do a proper back-end redirect, they will see this page for a second (or few pending their network latency).

also we'd just leave in the existing styling but change the contents; the header was the main thing I was thinking of leaving in (also favicon). we wouldn't maintain anything, that's how we'd leave it.
Although I guess that does get a little confusing once you dig into it... we'd remove everything in the header except for the tile and Argo icon, like the below:

Before:
![Screenshot 2023-12-20 at 5 15 07 PM](https://github.com/argoproj/argo-workflows/assets/4970083/8cd5c138-c645-4492-ad2f-a665086860a8)

After: 
![Screenshot 2023-12-20 at 5 15 07 PM - cropped](https://github.com/argoproj/argo-workflows/assets/4970083/08fb1c75-cd27-4bda-9862-9a0355f0a649)

I guess that doesn't really add much to the page altogether either now that I look at it... It is still a tad nicer, especially as the link text would be properly in the page and in the same font, as opposed to at the very top left of the page as it is with unstyled HTML.

> People should not be viewing the assets (css, js, etc) directly anyway IMO.

They shouldn't, but images in particular are one thing I can see people and bots/crawlers having links to that we may not want to break

Reply:
I'll see what I can do re: styling.

I'm feeling a bit dumb right now.  How do you redirect someone when they have a URL like `http://tinyurl.com/image.jpg`? Can you meta redirect the jpg endpoint?  I don't think so but I may learn today. 😄

Reply:
> I'll see what I can do re: styling.

I got the styling working in less than 100 LoC of HTML, let me share that shortly

> I'm feeling a bit dumb right now. How do you redirect someone when they have a URL like `http://tinyurl.com/image.jpg`? Can you meta redirect the jpg endpoint? I don't think so but I may learn today. 😄

Haha no I had the exact same thought process that the meta redirect makes that not possible since it only applies to HTML. I mean, you can do a meta redirect in a header as well, but we don't control the headers either

Reply:
HTML template below, replace `{{page_title}}` with the page title (e.g. "Contributing"), replace `{{page_subpath}}` with the subpath (e.g. `CONTRIBUTING`)

```html
<!doctype html>
<html lang="en" class="no-js">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <link rel="canonical" href="https://argo-workflows.readthedocs.io/en/stable/{{page_subpath}}/" />
    <meta http-equiv="refresh" content="0;URL=https://argo-workflows.readthedocs.io/en/stable/{{page_subpath}}/'" />
    <title>{{page_title}} - Argo Workflows - The workflow engine for Kubernetes</title>
    <link rel="icon" href="assets/images/favicon.png">
    <meta name="generator" content="mkdocs-1.2.4, mkdocs-material-8.1.9">

    <link rel="stylesheet" href="assets/stylesheets/main.2b4465f4.min.css">
    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
    <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
    <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
  </head>
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="">
      <header class="md-header md-header--lifted" data-md-component="header">
        <nav class="md-header__inner md-grid" aria-label="Header">
          <a href="." title="Argo Workflows - The workflow engine for Kubernetes" class="md-header__button md-logo" aria-label="Argo Workflows - The workflow engine for Kubernetes" data-md-component="logo">
            <img src="assets/logo.png" alt="logo">
          </a>
          <label class="md-header__button md-icon" for="__drawer">
            <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
          </label>
          <div class="md-header__title" data-md-component="header-title">
            <div class="md-header__ellipsis">
              <div class="md-header__topic">
                <span class="md-ellipsis">
                  Argo Workflows - The workflow engine for Kubernetes
                </span>
              </div>
            </div>
          </div>
        </nav>
      </header>

    <div class="md-container" data-md-component="container">
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
          </div>
          <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
          </div>
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">

              <h1>{{page_title}}</h1>
              <p>This page has moved to <a href="https://argo-workflows.readthedocs.io/en/latest/{{page_subpath}}/">https://argo-workflows.readthedocs.io/en/latest/{{page_subpath}}/</a>.</p>
              <p>You should be redirected there automatically. Please click the link above if you are not redirected.</p>

            </article>
          </div>
        </div>
      </main>

      <footer class="md-footer">
        <div class="md-footer-meta md-typeset">
          <div class="md-footer-meta__inner md-grid">
            <div class="md-copyright">
              Made with
              <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
                Material for MkDocs
              </a>
            </div>
          </div>
        </div>
      </footer>
    </div>
  </body>
</html>
```

Reply:
ok that HTML took me a bit to get working (and remove all unnecessary pieces and format correctly) so I might not mind being co-author on that one 😅

Reply:
Also with that HTML we can remove all the JS and the `palette` CSS as well (that was used for the light/dark theme)

Reply:
@agilgur5 you are now a co-author

- new styling in place
- had to dynamically determine assets path (index and 404 had no ../, nested index files have ../)
- javascript folder and palette CSS files are gone
- trimming of main.css is for later, like you suggested

Reply:
oh good catch on the pathing!